src/core/accumulators/Correlator.cpp

```diff
@@ -395,10 +393,9 @@ void Correlator::initialize() {
     }
 }
 
-int Correlator::get_data() {
+void Correlator::update() {
   if (finalized) {
     runtimeErrorMsg() << "No data can be added after finalize() was called.";
-    return 0;
```

Comment:
I think you have to keep the return statement, when you go on you might corrupt the data.